### PR TITLE
Allow send endpoint to persist state and send out the first question

### PR DESCRIPTION
**Pull request by chrisjluc**
https://github.com/uwblueprint/somo/issues/49

- Some concerns about what order messages will appear and how Twilio will split the messages if they're greater than 160 characters, but can be something to handle later.

### CODE REVIEW

File: lib/survey_execution/initializer.rb [hunk 1]
```diff
@@ -1,13 +1,19 @@
 module SurveyExecution
   class Initializer
-    def initialize(survey, respondents)
-      @client = Sms::Client.instance
+    def initialize(client, survey, respondents)
```
Comment (dinahshi) [inline lib/survey_execution/initializer.rb:3]: Why the refactor to take client as a parameter?

**Review by dinahshi — APPROVED** [review]
Looks good.

Would it make sense to wait for a response before sending the first question? Perhaps every `welcome_message` can end with something like: "If you think this was sent to you by mistake, reply STOP and we will stop sending you messages." Or even if the number is right but the respondent just doesn't want to participate. This would allow us to know that the user had seen the message and chose not to respond instead of us guessing what the status of the survey is.

File: lib/survey_execution/initializer.rb (hunk 1 above)
Comment (chrisjluc) [reply lib/survey_execution/initializer.rb:3]: for testing purposes

**Comment by chrisjluc** [comment]
That's a good idea, I'll add it to the backlog and we can prioritize it accordingly when the time comes. It will also involve some changes to the state machine to catch when the users sends a `STOP`

**Pull request merged**